pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -84,6 +86,32 @@ func (c *CSIControllerSet) WithManagementStateController(operandName string, sup
 	return c
 }
 
+// WithConditionalStaticResourcesController returns a *ControllerSet with a conditional static resources controller initialized.
+func (c *CSIControllerSet) WithConditionalStaticResourcesController(
+	name string,
+	kubeClient kubernetes.Interface,
+	dynamicClient dynamic.Interface,
+	kubeInformersForNamespace v1helpers.KubeInformersForNamespaces,
+	manifests resourceapply.AssetFunc,
+	files []string,
+	shouldCreateFnArg, shouldDeleteFnArg resourceapply.ConditionalFunction,
+) *CSIControllerSet {
+	c.conditionalStaticResourcesController = staticresourcecontroller.NewStaticResourceController(
+		name,
+		manifests,
+		[]string{},
+		(&resourceapply.ClientHolder{}).WithKubernetes(kubeClient).WithDynamicClient(dynamicClient),
+		c.operatorClient,
+		c.eventRecorder,
+	).WithConditionalResources(
+		manifests,
+		files,
+		shouldCreateFnArg,
+		shouldDeleteFnArg,
```

Comment:
Should we always supply an implicit `false` function here, rather than forcing callers to specify a dummy function which returns `false` ?

Reply:
IMO there are cases where users might not want to specify a dummy function here. I think it's worth leaving this open.